src/plugins/plot/configuration/PlotSeries.js

```diff
@@ -417,7 +430,11 @@ export default class PlotSeries extends Model {
         point.mctLimitState = this.evaluate(point);
         data.splice(insertIndex, 0, point);
         this.updateSeriesData(data);
-        this.emit('add', point, insertIndex, this);
+        // if user is not looking at data within the current bounds, don't notify chart of added point
```

Comment:
@shefalijoshi this probably needs your eyes. Per @akhenry's comment about `MctChart` firing `draw()` during fixed time mode, the issue appears to be we're triggering a `draw` when new telemetry comes into a series, even if that new telemetry falls outside the viewport. I've tried to avoid that below. Basically on viewport changes from `MctChart`, I set `xRangeMax`, `xRangeMin`, `yRangeMin`, and `yRangeMax` to what the user is looking at. If the incoming telemetry point is outside that range, I don't notify `MctChart` of the change. My worry is that others are listening to `add` beyond `MctChart` that need to know about data that falls outside the view port, but I don't see anything obvious. Thanks for the help!

Reply:
Thanks for the help @shefalijoshi! We moved the logic to `MctChart` wholly to avoid breaking the `add` event. You'll probably find this interesting @akhenry - before the fix with a 10hz telemetry source we get ~25% CPU usage in fixed time mode:

https://user-images.githubusercontent.com/9853862/213691948-876f7342-fb13-4aa6-a860-613edf2e67c0.mov

After the performance fix, we get 0.3%:

https://user-images.githubusercontent.com/9853862/213692004-16296894-eac6-406e-85f2-abaa4247a693.mov

Reply:
Love it! Thanks!